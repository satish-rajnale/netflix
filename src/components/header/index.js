import { useState } from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Profile,
  PlayButton,
  Picture,
  Dropdown,
  Group,
  Link,
  FeatureCallOut,
  Text,
  Background,
  ButtonLink,
  Container,
  Logo,
  Feature,
  SearchInput,
  SearchIcon,
  Search,
} from "./styles/header";
import React from 'react';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((searchActive) => !searchActive)}
      >
        <img src="/images/icons/search.png" alt="search" />
      </SearchIcon>
      <SearchInput
        placeholder="Search films & series"
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        active={searchActive}
      />
    </Search>
  );
};
