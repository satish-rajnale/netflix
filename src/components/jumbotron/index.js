import React from 'react';
import {Inner} from './styles/jumbotron'

export default function Jumbotron({children, direction = 'row', ...restProps}){
    return (
        <Inner direction={direction}>
            <p>Hello again</p>
        </Inner>
    )
}


Jumbotron.Container = function JumbotronContainer({ children, ...restProps}){
    return <Container {...restProps}>{children}</Container>
}