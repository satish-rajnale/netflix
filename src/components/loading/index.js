import { LockBody, Spinner, Picture, ReleaseBody } from "./styles/loading";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingReleasebody() {
  return <ReleaseBody />;
};
