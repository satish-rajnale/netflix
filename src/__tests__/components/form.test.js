import React from "react";
import { render} from "@testing-library/react";
import { Form } from "../../components";

jest.mock("react-router-dom")

describe("<Form/>", () => {
    it("renders <Form/> with data", () => {
        const {container, getByText, getByPlaceholderText} =render(
            <Form>
            <Form.Title>Sign In Now</Form.Title>
           
  
            <Form.Base >
              <Form.Input
                placeholder="Email Address"
              
                onChange={() => {}}
              />
              <Form.Input
                type="password"
                autoComplete="off"
                placeholder="Password"
               
                onChange={() => {}}
              />
              <Form.Submit disabled type="submit">
                Sign In
              </Form.Submit>
            </Form.Base>
            <Form.Text>
              New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
            </Form.Text>
            <Form.TextSmall>
              This page is protected by Google reCAPTCHA to ensure you're not a
              bot. Learn more.
            </Form.TextSmall>
          </Form>
        );

        expect(getByText("This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.")).toBeTruthy();
        expect(getByText("New to Netflix?")).toBeTruthy();
        expect(getByText("Sign In").disabled).toBeTruthy();
     
        expect(getByText("Sign In Now")).toBeTruthy();
        expect(getByPlaceholderText("Email Address")).toBeTruthy();
        expect(getByPlaceholderText("Password")).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();

    })
})