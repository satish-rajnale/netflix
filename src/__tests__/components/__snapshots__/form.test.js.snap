// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders <Form/> with data 1`] = `
<div
  class="form__Container-sc-7952rh-0 dEJqzD"
>
  <h1
    class="form__Title-sc-7952rh-3 fSOOxz"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-7952rh-2 ihDthY"
  >
    <input
      class="form__Input-sc-7952rh-7 dmqzSI"
      placeholder="Email Address"
    />
    <input
      autocomplete="off"
      class="form__Input-sc-7952rh-7 dmqzSI"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-7952rh-8 cqACVf"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-7952rh-4 iPwwaX"
  >
    New to Netflix? 
  </p>
  <p
    class="form__TextSmall-sc-7952rh-5 dNIyzB"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;

exports[`<Form/> renders <Form/>with error 1`] = `
<div
  class="form__Container-sc-7952rh-0 dEJqzD"
>
  <div
    class="form__Error-sc-7952rh-1 jQGvWO"
  >
    This email address is alredy registered.
  </div>
  <button
    class="form__Submit-sc-7952rh-8 cqACVf"
    type="submit"
  >
    Sign up
  </button>
</div>
`;
